src/Compilers/CSharp/Portable/Symbols/TypeUnification.cs

```diff
@@ -287,9 +287,10 @@ private static bool Contains(TypeSymbol type, TypeParameterSymbol typeParam)
                         NamedTypeSymbol namedType = (NamedTypeSymbol)type;
                         while ((object)namedType != null)
                         {
-                            foreach (TypeSymbol typeArg in namedType.TypeArgumentsNoUseSiteDiagnostics)
+                            ImmutableArray<TypeSymbol> typeParts = namedType.IsTupleType ? namedType.TupleElementTypes : namedType.TypeArgumentsNoUseSiteDiagnostics;
+                            foreach (TypeSymbol typePart in typeParts)
                             {
-                                if (Contains(typeArg, typeParam))
+                                if (Contains(typePart, typeParam))
```

Comment:
Can `TypeUnification.Contains` be replaced with the existing `TypeSymbolExtensions.ContainsTypeParameter`?